driver/driver.c

```diff
@@ -94,16 +98,55 @@ static const struct euicc_driver *_find_driver(enum euicc_driver_type type, cons
     return NULL;
 }
 
+int euicc_driver_list(int argc, char **argv) {
+    cJSON *payload = cJSON_CreateObject();
+    if (payload == NULL) return -1;
+
+    const struct euicc_driver *driver;
+    cJSON *name;
+    // APDU Drivers
+    {
+        cJSON *apdu_drivers = cJSON_CreateArray();
+        if (apdu_drivers == NULL) return -1;
+        for (int i = 0; drivers[i] != NULL; i++)
+        {
+            driver = drivers[i];
+            if (driver->type != DRIVER_APDU) continue;
+            name = cJSON_CreateString(driver->name);
+            if (name == NULL) return -1;
+            cJSON_AddItemToArray(apdu_drivers, name);
+        }
+        cJSON_AddItemToObject(payload, "LPAC_APDU", apdu_drivers);
+    }
+    // HTTP Drivers
+    {
+        cJSON *http_drivers = cJSON_CreateArray();
+        if (http_drivers == NULL) return -1;
+        for (int i = 0; drivers[i] != NULL; i++)
+        {
+            driver = drivers[i];
+            if (driver->type != DRIVER_HTTP) continue;
+            name = cJSON_CreateString(driver->name);
+            if (name == NULL) return -1;
+            cJSON_AddItemToArray(http_drivers, name);
+        }
+        cJSON_AddItemToObject(payload, "LPAC_HTTP", http_drivers);
+    }
```

Comment:
```suggestion
    cJSON *apdu_drivers = cJSON_CreateArray();
    cJSON *http_drivers = cJSON_CreateArray();
    if (apdu_drivers == NULL) return -1;
    for (int i = 0; drivers[i] != NULL; i++)
    {
        driver = drivers[i];
        name = cJSON_CreateString(driver->name);
        if (name == NULL) return -1;
        if (driver->type == DRIVER_APDU) {
            cJSON_AddItemToArray(apdu_drivers, name);
        } else if (driver->type == DRIVER_HTTP) {
            cJSON_AddItemToArray(http_drivers, name);
        }
    }
    cJSON_AddItemToObject(payload, "LPAC_APDU", apdu_drivers);
    cJSON_AddItemToObject(payload, "LPAC_HTTP", http_drivers);
    }
```
Just one loop is better.